### PR TITLE
test: demo MARC21 data from CDS

### PR DESCRIPTION
**Pull request by switowski**
- Adds test data from CERN Document Server.

### CODE REVIEW

**Comment by switowski** [comment]
TODO:
- [x] Clean data from CDS-specific tags

**Comment by switowski** [comment]
I have prepared an example record with fields that @SamiHiltunen was missing in https://github.com/inveniosoftware/dojson/issues/78.
@egabancho can you take a look and merge ?

File: dojson/contrib/marc21/fields/bd5xx.py [hunk 1]
```diff
@@ -123,10 +137,32 @@ def formatted_contents_note(self, key, value):
 def restrictions_on_access_note(self, key, value):
     """Restrictions on Access Note."""
     indicator_map1 = {
-        "#": "No information provided",
+        "_": "No information provided",
```
Comment (SamiHiltunen) [inline dojson/contrib/marc21/fields/bd5xx.py:140]: This can probably just be removed.

File: dojson/contrib/to_marc21/fields/bd01x09x.py [hunk 2]
```diff
@@ -1364,7 +1382,27 @@ def reverse_national_agricultural_library_copy_statement(self, key, value):
...
+        'subject_category_code_subdivision': 'x',
+        'source': '2',
+        'linkage': '6',
+        'field_link_and_sequence_number': '8',
+    }
+
+    order = utils.map_order(field_map, value)
+
+    if key[4] in indicator_map2:
+        order.append('code_source')
```
Comment (SamiHiltunen) [inline dojson/contrib/to_marc21/fields/bd01x09x.py:1404]: if key[4] == '7' we should order.remove('source')

File: dojson/contrib/to_marc21/fields/bd01x09x.py [hunk 3]
```diff
@@ -1435,7 +1473,27 @@ def reverse_dewey_decimal_classification_number(self, key, value):
...
+        'standard_or_optional_designation': 'm',
+        'assigning_agency': 'q',
+        'edition_number': '2',
+        'linkage': '6',
+        'field_link_and_sequence_number': '8',
+    }
+
+    order = utils.map_order(field_map, value)
+
+    if key[3] in indicator_map1:
```
Comment (SamiHiltunen) [inline dojson/contrib/to_marc21/fields/bd01x09x.py:1494]: edition_number should be removed from order if key[3] != '7'

File: dojson/contrib/to_marc21/fields/bd25x28x.py [hunk 4]
```diff
@@ -285,45 +285,88 @@ def reverse_production_publication_distribution_manufacture_and_copyright_notice
...
+        'field_link_and_sequence_number': '8',
+    }
+
+    order = utils.map_order(field_map, value)
+
+    if key[3] in indicator_map1:
+        order.append('level')
+
+    if key[4] in indicator_map2:
+        order.append('type_of_address')
```
Comment (SamiHiltunen) [inline dojson/contrib/to_marc21/fields/bd25x28x.py:326]: type_of_address can be removed from order if key[4] != '7'

File: dojson/contrib/to_marc21/fields/bd25x28x.py [hunk 5]
```diff
@@ -285,45 +285,88 @@ def reverse_production_publication_distribution_manufacture_and_copyright_notice
...
-        'g': value.get('attention_name'),
+        'e': value.get('postal_code'),
         'f': value.get('terms_preceding_attention_name'),
-        'i': value.get('type_of_address'),
+        'g': value.get('attention_name'),
         'h': value.get('attention_position'),
-        'k': utils.reverse_force_list(
-            value.get('telephone_number')
-        ),
+        'i': value.get('type_of_address'),
```
Comment (SamiHiltunen) [inline dojson/contrib/to_marc21/fields/bd25x28x.py]: remove this duplicate 'i' key, it is being set at the bottom of this return correctly

File: dojson/contrib/to_marc21/fields/bd5xx.py [hunk 6]
```diff
@@ -134,7 +148,29 @@ def reverse_restrictions_on_access_note(self, key, value):
         "No information provided": "_",
```
Comment (SamiHiltunen) [inline dojson/contrib/to_marc21/fields/bd5xx.py:148]: this can be removed

**Comment by switowski** [comment]
Thanks for spotting all those errors @SamiHiltunen !
As for removing those 'no information provided' and similar indicators, I wouldn't do this. Not always `_` has an obvious meaning, for example here: https://www.loc.gov/marc/bibliographic/concise/bd210.html it actually have a completely different meaning.

File: dojson/contrib/to_marc21/fields/bd01x09x.py (hunk 2 above)
Comment (switowski) [inline dojson/contrib/to_marc21/fields/bd01x09x.py:1404]: Something like that (I guess you mean if key[4] **!=** '7'):

```
@to_marc21.over('072', '^subject_category_code$')
@utils.reverse_for_each_value
@utils.filter_values
def reverse_subject_category_code(self, key, value):
    """Reverse - Subject Category Code."""

    indicator_map2 = {
        "Source specified in subfield $2": "7",
        "NAL subject category code list ": "0"
    }

    field_map = {
        'subject_category_code': 'a',
        'subject_category_code_subdivision': 'x',
        'source': '2',
        'linkage': '6',
        'field_link_and_sequence_number': '8',
    }

    order = utils.map_order(field_map, value)

    if key[4] in indicator_map2:
        order.append('code_source')

    if key[4] != '7':
        order.remove('2')

    return {
        '__order__': tuple(order) if len(order) else None,
        'a': value.get('subject_category_code'),
        '8': utils.reverse_force_list(
            value.get('field_link_and_sequence_number')
        ),
        '2': value.get('source'),
        'x': utils.reverse_force_list(
            value.get('subject_category_code_subdivision')
        ),
        '6': value.get('linkage'),
        '$ind1': '_',
        '$ind2': indicator_map2.get(value.get('code_source'), '_'),
    }
```

(double checking cause the more I work with it, the more I get confused)

Comment (SamiHiltunen) [inline dojson/contrib/to_marc21/fields/bd01x09x.py:1404]: Yep, that is what I meant.

**Comment by tiborsimko** [comment]
Note that this commit updates field rules, but does not update schemas.  The danger is that they may get out of sync...

```
dojson/contrib/marc21/fields/bd3xx.py
dojson/contrib/marc21/schemas/bd3xx.json
```

**Comment by switowski** [comment]
Valid point. https://github.com/inveniosoftware/dojson/issues/84 should catch those issues in the future.

**Pull request merged**